### PR TITLE
workflows support reuse id policy

### PR DESCRIPTION
**Pull request by famarting**
# Description

Implements the missing bits to support workflows reuse id policy

## Issue reference

Please reference the issue this PR will close: #493

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Review by mikeee — CHANGES_REQUESTED** [review]
I think surfacing this struct as an option is quite confusing in its current form, could the function possibly be refactored to not require another (internal) struct from `microsoft/durabletask-go` and/or document this before merge?

**Comment by famarting** [comment]
> I think surfacing this struct as an option is quite confusing in its current form, could the function possibly be refactored to not require another (internal) struct from `microsoft/durabletask-go` and/or document this before merge?

done

File: workflow/client.go [hunk 1]
```diff
@@ -31,6 +31,11 @@ type Client struct {
 	taskHubClient *durabletaskclient.TaskHubGrpcClient
 }
 
+type WorkflowIDReusePolicy struct {
+	OperationStatus []api.OrchestrationStatus
+	Action          api.CreateOrchestrationAction
```
Comment (mikeee) [inline workflow/client.go:36]: Could these types be defined/wrapped as part of the go-sdk and parsed in the options function below?

**Review by mikeee — APPROVED** [review]

**Comment by famarting** [comment]
merge please

**Pull request merged**